src/core/stringHelper.ml

```diff
@@ -19,7 +19,7 @@ let capitalize s =
 		Bytes.to_string bytes
 
 let starts_uppercase x =
```

Comment:
Maybe we should rename this to something that indicates that it checks for valid type names. It's strange to have something called `starts_uppercase` accept `_` as I don't think that's commonly considered as an upper-case character.

Reply:
Hm.. "type name" is kind of misleading too, since we need it for module names as well which have the same naming semantics. That's why I went with `check_uppercase_identifier_name`  in `Typecore`.

Reply:
There's also a `check_uppercase` right after this, which sounds generic since it's in `StringHelper`, but in reality it throws a "Class name must start with uppercase character" error...